Crash on Android 5-7 because of dependency on java.util.Base64 in Gist upgraded in 3.3.0

<!-- For time-sensitive or urgent requests, reach out to our Technical Support team at win@customer.io with the details below instead of submitting an issue here.  -->

**SDK version:** 3.3.0 - 3.4.2

**Environment:** Production

**Are logs available?** 
Error fetching messages: canceled due to java.lang.NoClassDefFoundError: Failed resolution of: Ljava/util/Base64;
2023-05-06 17:05:04.673 22480-22563 AndroidRuntime          cz.ai.aschool.movies                 E  FATAL EXCEPTION: OkHttp Dispatcher
                                                                                                    Process: cz.ai.aschool.movies, PID: 22480
                                                                                                    java.lang.NoClassDefFoundError: Failed resolution of: Ljava/util/Base64;
                                                                                                    	at build.gist.data.listeners.Queue$gistQueueService$2.invoke$lambda-2(Queue.kt:35)
                                                                                                    	at build.gist.data.listeners.Queue$gistQueueService$2.$r8$lambda$PAgpBkRd0H90geyu2KeSV3WZpLE(Queue.kt)
                                                                                                    	at build.gist.data.listeners.Queue$gistQueueService$2$$ExternalSyntheticLambda0.intercept(D8$$SyntheticClass)
                                                                                                    	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
                                                                                                    	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
                                                                                                    	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                                                                                                    	at java.lang.Thread.run(Thread.java:818)
                                                                                                    Caused by: java.lang.ClassNotFoundException: Didn't find class "java.util.Base64" on path: DexPathList[[zip file "/data/app/cz.ai.aschool.movies-2/base.apk"],nativeLibraryDirectories=[/data/app/cz.ai.aschool.movies-2/lib/x86, /data/app/cz.ai.aschool.movies-2/base.apk!/lib/x86, /vendor/lib, /system/lib]]
                                                                                                    	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
                                                                                                    	at build.gist.data.listeners.Queue$gistQueueService$2.invoke$lambda-2(Queue.kt:35) 
                                                                                                    	at build.gist.data.listeners.Queue$gistQueueService$2.$r8$lambda$PAgpBkRd0H90geyu2KeSV3WZpLE(Queue.kt) 
                                                                                                    	at build.gist.data.listeners.Queue$gistQueueService$2$$ExternalSyntheticLambda0.intercept(D8$$SyntheticClass) 
                                                                                                    	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109) 
                                                                                                    	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201) 
                                                                                                    	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517) 
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
                                                                                                    	at java.lang.Thread.run(Thread.java:818) 
                                                                                                    	Suppressed: java.lang.ClassNotFoundException: java.util.Base64
                                                                                                    		at java.lang.Class.classForName(Native Method)
                                                                                                    		at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
                                                                                                    		at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
                                                                                                    		at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
                                                                                                    		... 10 more
                                                                                                    	Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
2023-05-06 17:05:08.756 22480-22554 Gist                    cz.ai.aschool.movies                 E  Error fetching messages: canceled due to java.lang.NoClassDefFoundError: Failed resolution of: Ljava/util/Base64;

**Describe the bug**
After updating to version 3.3.2 I have found, that my app starts crashing. Last working cio SKD version is 3.2.0.
I suspect Gist dependency update as a root cause of the problem, as java.util.Base64 is available from Android 8.
But cio declares support for Android 5 so this is why i consider it to be a bug.

**To Reproduce**
Run your SDK on Android 5-7

**Expected behavior**
<!--A clear and concise description of what you expected to happen.-->

**Screenshots**
<!--If applicable, add screenshots to help explain your problem.-->

**Additional context**
<!--Add any other context about the problem here, including links to any campaigns, broadcasts, or profiles related to this bug. If it's required or relevant, include code! -->

--- a/gist/build.gradle
+++ b/gist/build.gradle
@@ -6,7 +6,7 @@ android {
     compileSdkVersion 31
     buildToolsVersion "30.0.2"
     defaultConfig {
-        minSdkVersion 21
+        minSdkVersion 19
         targetSdkVersion 31
 
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"

--- a/gist/build.gradle
+++ b/gist/build.gradle
@@ -3,11 +3,11 @@ apply plugin: 'kotlin-android'
 apply plugin: 'maven-publish'
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 31
     buildToolsVersion "30.0.2"
     defaultConfig {
-        minSdkVersion 19
-        targetSdkVersion 30
+        minSdkVersion 21
+        targetSdkVersion 31
 
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         consumerProguardFiles 'consumer-rules.pro'

--- a/gist/src/main/java/build/gist/data/listeners/Queue.kt
+++ b/gist/src/main/java/build/gist/data/listeners/Queue.kt
@@ -1,5 +1,6 @@
 package build.gist.data.listeners
 
+import android.util.Base64
 import android.util.Log
 import build.gist.data.NetworkUtilities
 import build.gist.data.model.GistMessageProperties
@@ -14,10 +15,9 @@ import okhttp3.OkHttpClient
 import okhttp3.Request
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
-import java.util.Base64
 import java.util.regex.PatternSyntaxException
 
-class Queue: GistListener {
+class Queue : GistListener {
 
     private var localMessageStore: MutableList<Message> = mutableListOf()
 
@@ -32,7 +32,11 @@ class Queue: GistListener {
                     val request: Request = chain.request().newBuilder()
                         .addHeader(NetworkUtilities.CIO_SITE_ID_HEADER, GistSdk.siteId)
                         .addHeader(NetworkUtilities.CIO_DATACENTER_HEADER, GistSdk.dataCenter)
-                        .addHeader(NetworkUtilities.USER_TOKEN_HEADER, Base64.getEncoder().encodeToString(userToken.toByteArray()))
+                        .addHeader(
+                            NetworkUtilities.USER_TOKEN_HEADER,
+                            // The NO_WRAP flag will omit all line terminators (i.e., the output will be on one long line).
+                            Base64.encodeToString(userToken.toByteArray(), Base64.NO_WRAP)
+                        )
                         .build()
 
                     chain.proceed(request)
@@ -72,11 +76,13 @@ class Queue: GistListener {
                     // No content, don't do anything
                     Log.i(GIST_TAG, "No messages found for user")
                 } else if (latestMessagesResponse.isSuccessful) {
-                    Log.i(GIST_TAG, "Found ${latestMessagesResponse.body()?.count()} messages for user")
+                    Log.i(
+                        GIST_TAG,
+                        "Found ${latestMessagesResponse.body()?.count()} messages for user"
+                    )
                     latestMessagesResponse.body()?.let { handleMessages(it) }
                 }
-            }
-            catch (e: Exception) {
+            } catch (e: Exception) {
                 Log.e(
                     GIST_TAG,
                     "Error fetching messages: ${e.message}"
@@ -126,7 +132,10 @@ class Queue: GistListener {
         GlobalScope.launch {
             try {
                 if (message.queueId != null) {
-                    Log.i(GIST_TAG, "Logging view for user message: ${message.messageId}, with queue id: ${message.queueId}")
+                    Log.i(
+                        GIST_TAG,
+                        "Logging view for user message: ${message.messageId}, with queue id: ${message.queueId}"
+                    )
                     removeMessageFromLocalStore(message)
                     gistQueueService.logUserMessageView(message.queueId)
                 } else {
@@ -140,8 +149,11 @@ class Queue: GistListener {
     }
 
     private fun addMessageToLocalStore(message: Message) {
-        val localMessage = localMessageStore.find { localMessage -> localMessage.queueId == message.queueId }
-        if (localMessage == null) { localMessageStore.add(message) }
+        val localMessage =
+            localMessageStore.find { localMessage -> localMessage.queueId == message.queueId }
+        if (localMessage == null) {
+            localMessageStore.add(message)
+        }
     }
 
     private fun removeMessageFromLocalStore(message: Message) {

